modules/xfem/include/dirackernels/XFEMPressure.h

```diff
@@ -0,0 +1,37 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#ifndef XFEMPRESSURE_H
+#define XFEMPRESSURE_H
+
+// Moose Includes
+#include "DiracKernel.h"
+
+class Function;
+
+class XFEMPressure : public DiracKernel
+{
+public:
+  XFEMPressure(const InputParameters & parameters);
+
+  virtual void addPoints();
+  virtual Real computeQpResidual();
+
+protected:
+  const int _component;
+  const Real _factor;
+  Function * const _function;
+  bool _displaced;
+
+  std::map<unsigned int, MooseSharedPointer<ElementPairLocator> > * element_pair_locators;
+  std::map<const Elem*, std::map<Point, Point> > _elem_point_normal;
+  std::map<const Elem*, std::map<Point, Real> > _elem_point_JxW;
```

Comment:
Find an alternate design to `std::map<Point, X>` if possible!  The `Point::operator<` is highly susceptible to floating point differences, and shouldn't be used for this.